modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -389,6 +390,81 @@ public void testWalWrittenBytes() throws Exception {
         assertEquals(exp, ((LongAdderMetric)dsMetricRegistry(n).findMetric("WalWrittenBytes")).value());
     }
 
+    /**
+     * Check whether Wal is reporting correct usage.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testWalTotalSize() throws Exception {
+        IgniteEx n = startGrid(0, (Consumer<IgniteConfiguration>)cfg ->
+            cfg.getDataStorageConfiguration().setWalSegmentSize((int)(2 * U.MB)));
+
+        n.cluster().state(ACTIVE);
+        awaitPartitionMapExchange();
+
+        for (int i = 0; i < 10; i++)
+            n.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+
+        long exp = 10 * 2048 * 1024;
```

Comment:
At this moment, you need to turn off the recording in WAL, as it is done here:
https://github.com/gridgain/gridgain/blob/7ca6a4f2afdd5ae51e8d420770f7e8a55ae89a6e/modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java#L376

Then calculate the size of the WAL archive using the code like this:
```
        FileWriteAheadLogManager walMgr = walMgr(n);
        SegmentRouter router = walMgr.getSegmentRouter();

        long totalSize = walMgr.totalSize(FileWriteAheadLogManager.loadFileDescriptors(router.getWalWorkDir()));

        if (router.hasArchive())
            totalSize += walMgr.totalSize(FileWriteAheadLogManager.loadFileDescriptors(router.getWalArchiveDir()));

```